### ISSUE TITLE
Errors

### ISSUE THREAD

**Issue by lorrp1** [jdb78/pytorch-forecasting#8]
Hello, im stallion.py but im getting few problems:
using gpu = 1 in the trainer return
>  models/temporal_fusion_transformer/__init__.py", line 793, in _log_interpretation dim=0

using only the cpu it works but return this error: 
>  AttributeError: module 'tensorflow._api.v1.io.gfile' has no attribute 'get_filesystem'
which apparently it is due to pytorch and tensorboard incompatibility but i managed to install pytorch-forecasting without problems so could you tell me what version of pythorch/tensorboard are you using?

thanks

**Comment by jdb78** [comment]
This is a known tensorboard issue. Deinstall tensorflow and then install tensorboard 2.2 (I think 2.3 has an unfixed bug).

On GPU: I have not tested that recently. If you know of any free CI with GPUs, I would be very keen to learn about it.

**Comment by lorrp1** [comment]
using a way smaller bench size return a way better result.  
i have seen the "fix gpu", im going to check soon if it works now.

**Comment by jdb78** [comment]
Have you checked that the training works on GPU for you? I am always grateful for feedback!

**Comment by lorrp1** [comment]
I haven’t got time.
But I’ll try as soon as possible.

**Comment by lorrp1** [comment]
not working with the last update from pip :

> .local/lib/python3.7/site-packages/pytorch_forecasting/models/temporal_fusion_transformer/__init__.py", line 447, in forward
    input_vectors[name] = emb(x_cat[..., self.hparams.x_categoricals.index(name)])
....
> .local/lib/python3.7/site-packages/torch/nn/functional.py",` line 1814, in embedding

    return torch.embedding(weight, input, padding_idx, scale_grad_by_freq, sparse)
RuntimeError: Expected object of device type cuda but got device type cpu for argument #3 'index' in call to _th_index_select

**Comment by jdb78** [comment]
Thanks for the bug - have not run into this yet. Interesting that this seems to happen after 16 epochs. It's definitely time to employ GPUs for testing - I will give this a shot next week.

**Comment by jdb78** [comment]
@lorrp1 Are you sure to be on the newest version. The line 447 the traceback cites seems to be 449 on the master (https://github.com/jdb78/pytorch-forecasting/blob/master/pytorch_forecasting/models/temporal_fusion_transformer/__init__.py#L449)

**Comment by lorrp1** [comment]
by "last from pip" i meant to write pypi.
with the last from the master (0.2.4) same: 
```

Epoch 48: 100%|███████████| 27Traceback (most recent call last):1.221, v_num=56]
  File "/home/klop/Documents/trading/TimeSeries/Multivariate/pytorchForecasting/randomTIme/stallion2.py", line 113, in <module>
    preds, index = tft.predict(val_dataloader, return_index=True, fast_dev_run=True)
  File "/home/klop/.local/lib/python3.7/site-packages/pytorch_forecasting/models/base_model.py", line 541, in predict
    out = self(x)  # raw output is dictionary
  File "/home/klop/.local/lib/python3.7/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/klop/.local/lib/python3.7/site-packages/pytorch_forecasting/models/temporal_fusion_transformer/__init__.py", line 449, in forward
    input_vectors[name] = emb(x_cat[..., self.hparams.x_categoricals.index(name)])
  File "/home/klop/.local/lib/python3.7/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/klop/.local/lib/python3.7/site-packages/torch/nn/modules/sparse.py", line 126, in forward
    self.norm_type, self.scale_grad_by_freq, self.sparse)
  File "/home/klop/.local/lib/python3.7/site-packages/torch/nn/functional.py", line 1814, in embedding
    return torch.embedding(weight, input, padding_idx, scale_grad_by_freq, sparse)
RuntimeError: Expected object of device type cuda but got device type cpu for argument #3 'index' in call to _th_index_select
```
also its pretty strange that the gpu=0 version ended at the 32 epoch (without errors)

**Comment by jdb78** [comment]
I think the finishing without errors is due to early stopping.

**Comment by jdb78** [comment]
Fixed in #27

**Comment by lorrp1** [comment]
it works adding tft.to("cpu") before .predict
